### PR TITLE
JavaScript capstone project - API-based webapp

### PR DESCRIPTION
**Pull request by Ghiftee**
In this project, my partner @tacettinkustu and I created a webapp using a TVmaze API that gives data about TV series and Movies. The TVmaze API allowed us to get a list of items with a unique ID and to also get detailed information about the item.
In addition to the TVmaze API, we used an Involvement API created by Microverse to record interactions on the app such as likes and comments.

Using the wireframe provided, we built a header, footer, homepage and comments popup page.

Our project makes use of the following:

- ES6 syntax
- ES6 modules
- Callback and promises
- Webpack
- AAA pattern for unit testing using Jest
- Gitflow

I worked as Student A on this project and I implemented everything that has to do with the movies display on the homepage and the like button.
@tacettinkustu worked as Student B and he created everything that involves the comment button and comment page popup.

Each branch on this project was created from the dev branch and they represent a card on the Kanban board. After working on each card/branch(excluding the set-up branch which we worked on together), my partner and I reviewed each other's codes and did not begin work on the next card/branch until our pull request was approved.

Kindly review our code.
Thank you!

### CODE REVIEW

File: src/api/involvement.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+const getLikes = async () => {
+  const appId = localStorage.getItem('appId');
+  const response = await fetch(`${baseUrl}/apps/${appId}/likes`);
+  let likes = await response.text();
+  if (likes.length === 0) likes = '[]';
+  return JSON.parse(`{"likes": ${likes}}`);
+};
+
+module.exports = { createApp, likeMovie, getLikes };
```
Comment (ZawadiSumba66) [inline src/api/involvement.js]: - [x] Seems you forgot to apply ES6 syntax here on exports. Kindly fix this.

File: src/comments/popup.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+};
+
+export const getCounts = (arr) => arr.length;
+
+const incrementCount = (title, commentsInfo) => {
+  const value = getCounts(commentsInfo) + 1;
+  title.textContent = `Comments (${value})`;
+};
+
+export default async function gotoCommentPage(showId) {
```
Comment (ZawadiSumba66) [inline src/comments/popup.js]: - [x] Another reminder that the function above  should be an arrow function while following the ES6 rules.

**Review by ZawadiSumba66 — CHANGES_REQUESTED** [review]
### Hi Team 😃 , 

Good job so far!
### HIGHLIGHTS 🚀 
- Video Presentation is clear ✔️ 
- Project UI is amazing ✔️ 
- Descriptive README file ✔️ 
- Mastery of user interactions with the usage of Involvement API. ✔️ 

However, there are some issues on ES6 syntax that you still need to work on to go to the next step but you are almost done!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/api/involvement.js (hunk 1 above)
Comment (tacettinkustu) [reply src/api/involvement.js]: We applied ES6 syntax for export functions.

File: src/comments/popup.js (hunk 2 above)
Comment (tacettinkustu) [reply src/comments/popup.js]: We replaced function with an arrow function while following the ES6 rules in popup file.

**Comment by tacettinkustu** [comment]
> ### Hi Team 😃 ,
> Good job so far!
> 
> ### HIGHLIGHTS 🚀
> * Video Presentation is clear ✔️
> * Project UI is amazing ✔️
> * Descriptive README file ✔️
> * Mastery of user interactions with the usage of Involvement API. ✔️
> 
> However, there are some issues on ES6 syntax that you still need to work on to go to the next step but you are almost done!
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Thank you for your insightful reviews.

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @Ghiftee &  @tacettinkustu, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**